iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/string.inl

```diff
@@ -639,16 +567,17 @@ inline IsStringOrCharArray<T, optional<uint64_t>> string<Capacity>::find_last_of
 template <uint64_t Capacity>
 inline constexpr char& string<Capacity>::at(const uint64_t pos) noexcept
 {
-    // const_cast to avoid code duplication, safe since it's first casted to a const type and then the const is removed
-    // again
+    // AXIVION Next Construct AutosarC++19_03-A5.2.3 : const_cast to avoid code duplication, safe since it's first
+    // casted to a const type and then the const is removed
     // NOLINTNEXTLINE(cppcoreguidelines-pro-type-const-cast)
     return const_cast<char&>(const_cast<const string<Capacity>*>(this)->at(pos));
 }
 
 template <uint64_t Capacity>
 inline constexpr const char& string<Capacity>::at(const uint64_t pos) const noexcept
 {
-    Expects(pos < size() && "Out of bounds access!");
+    // AXIVION Next Construct AutosarC++19_03-M5.3.1 : operand of logical operator is non-bool, enhances readability
+    Expects((pos < size()) && ("Out of bounds access!"));
```

Comment:
You could replace this line with
```suggestion
    ExpectsWithMsg((pos < size()), "Out of bounds access!");
```
Then you need no suppression and the test coverage will also increase.